### PR TITLE
feat(commitlint): add commitlint linter to follow conventional commit specifications

### PR DESCRIPTION
**Pull request by pavanjoshi914**
Fixes:  https://github.com/getAlby/lightning-browser-extension/issues/612
add commitment linter
setup pre-commit hook ```commit-msg`` using husky
update readme

when commit message doesn't follow commit conventions

when commit message follows commit conventions

### CODE REVIEW

**Comment by pavanjoshi914** [comment]
@bumi @escapedcat kindly review

https://github.com/apps/commit-lint  I also found this app, which when integrate, check for commit conventions on commits, on GitHub itself instead of doing it locally.


once the pr is created, it will start checks on that PR

image
image

File: README.md
Comment (escapedcat) [inline README.md:158]: No need to link to a specific version I guess, just `https://www.conventionalcommits.org` is fine.

Comment (escapedcat) [reply README.md:158]: Personally this would be enough for me but some projects explain the main things in a short summary. Let's see what the others think.

Comment (pavanjoshi914) [reply README.md:158]: There's a lot of stuff out there in the readme so kept it short. we can create a new markdown named Contributing Guidelines instead. WDYT?

Comment (escapedcat) [reply README.md:158]: Good idea, but no need for now I think. Let's see what others think.  
These guides can be very helpful, i.e. https://github.com/LN-Zap/zap-desktop/blob/master/docs/CONTRIBUTING.md

**Comment by bumi** [comment]
I think @escapedcat should decide this. He has the best knowledge of this. ;) 
Contributing should just stay as easy and open as possible.

**Comment by escapedcat** [comment]
> Contributing should just stay as easy and open as possible

Agreed. I don't see an issue with that. But I think the team should stand behind such decision.  

@bumi Have a look at this: https://github.com/getAlby/lightning-browser-extension/issues/612#issuecomment-1045966986  
Happy to talk about any concerns before we merge this.

**Comment by pavanjoshi914** [comment]
> > Contributing should just stay as easy and open as possible
> 
> Agreed. I don't see an issue with that. But I think the team should stand behind such decision.
> 
> @bumi Have a look at this: [#612 (comment)](https://github.com/getAlby/lightning-browser-extension/issues/612#issuecomment-1045966986) Happy to talk about any concerns before we merge this.

@escapedcat shall we add pull request template as well.
Also what you think about integrating commit-lint bot which will automate lint check at GitHub level.

This will also be a solution when user use --no-verify flag to bypass commitlint locally

**Comment by escapedcat** [comment]
> @escapedcat shall we add pull request template as well. Also what you think about integrating commit-lint bot as mentioned above which will automate lint check at GitHub level.

> This will also be a solution when user use --no-verify flag to bypass commitlint locally  

I think it would be fine to first see how the local hook works for everyone and if we gain something from linted commits, i.e. semver. But it would be great next step if everyone is happy with linting.

**Comment by pavanjoshi914** [comment]
> > @escapedcat shall we add pull request template as well. Also what you think about integrating commit-lint bot as mentioned above which will automate lint check at GitHub level.
> 
> > This will also be a solution when user use --no-verify flag to bypass commitlint locally
> 
> I think it would be fine to first see how the local hook works for everyone and if we gain something from linted commits, i.e. semver. But it would be great next step if everyone is happy with linting.

Agreed! If this gets merged, we can make a announcement to try and give feedbacks if everyone is comfortable with local hooks.

Else we can add  lint check after creating the pr on github through bot mentioned above

**Comment by pavanjoshi914** [comment]
https://github.com/semantic-release/semantic-release
https://github.com/conventional-changelog/standard-version

Automated semver when conventional commit specifications are applied

File: .husky/commit-msg
Comment (dylancom) [inline .husky/commit-msg:4]: `npx --no -- commitlint --edit $1`

Comment (escapedcat) [reply .husky/commit-msg:4]: What's the reason for this? Both should work fine.

Comment (pavanjoshi914) [reply .husky/commit-msg:4]: In the readme, we specify ```yarn``` as a package manager. so preferred yarn to add and run commit-msg hook locally

Comment (escapedcat) [reply .husky/commit-msg:4]: I'm good with either. Just wanted to know the reasoning behind this to understand the motivation.

Comment (pavanjoshi914) [reply .husky/commit-msg:4]: any updates here @bumi @escapedcat. I was willing to get this pr merged so that newer changes in readme doesn't create any merge conflicts.

Comment (bumi) [reply .husky/commit-msg:4]: I have no idea. I trust you. @dylancom why do you recommend `npx`?

**Review by dylancom — APPROVED** [review]

File: .husky/commit-msg
Comment (dylancom) [reply .husky/commit-msg:4]: Both should work fine indeed, my comment was just about consistency. Let's update our pre-commit hook with yarn too then :)

Comment (pavanjoshi914) [reply .husky/commit-msg:4]: shall I change to npx then?. but in readme, we specify yarn as a package manager. we shall also mention there npx as well. if user use older npm version(they don't contain npx by default) or somehow npx doesnt work in system then it might be a problem.

Comment (dylancom) [reply .husky/commit-msg:4]: Let's go for Yarn, would you mind updating .husky/pre-commit?

Comment (pavanjoshi914) [reply .husky/commit-msg:4]: okay, I will update this in new PR!

Comment (pavanjoshi914) [reply .husky/commit-msg:4]: @bumi  kindly review and merge

**Pull request merged**
